libs/contrib/Data/Rel/Extra.idr

```diff
@@ -0,0 +1,37 @@
+module Data.Rel.Extra
+
+import Data.Rel
+import Data.Fun
+import Data.Fun.Extra
+import Data.HVect
+
+%default total
+
+||| The logical complement of a relation.
+public export
+complement : {ts : Vect n Type} -> (p : Rel ts) -> Rel ts
```

Comment:
This is an instance of [`map`](https://github.com/JoeyEremondi/Idris2/blob/finiteDec/libs/base/Data/Fun.idr#L18)